scene/2d/canvas_item.cpp

```diff
@@ -263,7 +263,8 @@ bool CanvasItem::is_visible_in_tree() const {
 
 void CanvasItem::_propagate_visibility_changed(bool p_visible) {
 
-	notification(NOTIFICATION_VISIBILITY_CHANGED);
+	if (!first_draw)
+		notification(NOTIFICATION_VISIBILITY_CHANGED);
```

Comment:
Given that `first_draw` is only used to trigger this notification, doesn't it mean that it's just not needed?

Reply:
I just beware it could break some usages, but I could remove it instead this check of course..

Reply:
Yeah I don't know either if removing it would be safe.

Reply:
Both occurrences of the notification come from the original open sourcing commit of Godot, so can't check the git history further up to see why it's done this way. CC @reduz

Reply:
(After test with simple label) No it should not be removed from canvas_item unless you want break backward compatibility.

Reply:
Thanks for checking :)